src/CoreBundle/Bridge/Chameleon/Module/NavigationTreeSingleSelect/NavigationTreeSingleSelect.php

```diff
@@ -0,0 +1,173 @@
+use ChameleonSystem\CoreBundle\Factory\BackendTreeNodeFactory;
+use ChameleonSystem\CoreBundle\ServiceLocator;
+use ChameleonSystem\CoreBundle\Util\InputFilterUtilInterface;
+use Doctrine\DBAL\Connection;
+use MTPkgViewRendererAbstractModuleMapper;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use TGlobal;
+
+class NavigationTreeSingleSelect extends MTPkgViewRendererAbstractModuleMapper
+{
+    /**
+     * @var Connection
+     */
+    private $dbConnection;
+    /**
+     * @var EventDispatcherInterface
+     */
+    private $eventDispatcher;
+    /**
+     * @var InputFilterUtilInterface
+     */
+    private $inputFilterUtil;
+    /**
+     * @var BackendTreeNodeFactory
+     */
+    private $backendTreeNodeFactory;
+    /**
+     * @var string
+     */
+    private $treePathHTML = '';
+
+    /**
+     * Nodes that should not be assignable or that should have only a
+     * restricted context menu.
+     *
+     * @var array
+     */
+    protected $restrictedNodes = [];
+
+    public function __construct(
+        Connection $dbConnection,
+        EventDispatcherInterface $eventDispatcher,
+        InputFilterUtilInterface $inputFilterUtil,
+        BackendTreeNodeFactory $backendTreeNodeFactory
+    ) {
+        $this->dbConnection = $dbConnection;
+        $this->eventDispatcher = $eventDispatcher;
+        $this->inputFilterUtil = $inputFilterUtil;
+        $this->backendTreeNodeFactory = $backendTreeNodeFactory;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function Accept(\IMapperVisitorRestricted $visitor, $cachingEnabled, \IMapperCacheTriggerRestricted $cacheTriggerManager)
+    {
+        $fieldName = $this->inputFilterUtil->getFilteredGetInput('fieldName', '');
+        $nodeID = $this->inputFilterUtil->getFilteredGetInput('id', '');
+
+        $this->restrictedNodes = $this->getPortalNavigationStartNodes();
+
+        $rootTreeId = $this->getPortalTreeRootNodeId();
+        if ('' !== $rootTreeId) {
+            $rootNode = new \TdbCmsTree();
+            $rootNode->SetLanguage(\TdbCmsUser::GetActiveUser()->GetCurrentEditLanguageID());
+            $rootNode->Load($rootTreeId);
+
+            $treeNodes = $this->createTreeDataModel($rootNode, $fieldName);
+            $visitor->SetMappedValue('treeNodes', $treeNodes);
+        }
+        $visitor->SetMappedValue('treePathHTML', $this->treePathHTML);
+        $visitor->SetMappedValue('activeId', $nodeID);
+        $visitor->SetMappedValue('fieldName', $fieldName);
+        $visitor->SetMappedValue('level', 0);
+    }
+
+    private function getPortalTreeRootNodeId(): string
+    {
+        $portalId = $this->inputFilterUtil->getFilteredGetInput('portalID', '');
+
+        if ('' === $portalId) {
+            return \TCMSTreeNode::TREE_ROOT_ID;
+        }
+
+        $portal = new \TCMSPortal();
+        $portal->Load($portalId);
+        if (null === $portal || '' === $portal->sqlData['main_node_tree']) {
```

Comment:
using the tdb object you should use the $portal->fieldMainNodeTree property instead of sqlData

Reply:
ok, done